src/pushsource/_impl/backend/staged/staged_directory.py

```diff
@@ -0,0 +1,18 @@
+import logging
+
+from ...model import DirectoryPushItem
+from .staged_base import StagedBaseMixin, handles_type
+
+LOG = logging.getLogger("pushsource")
+
+
+class StagedDirectoryMixin(StagedBaseMixin):
+    @handles_type("RAW")
+    # pylint: disable=unused-argument
+    def __directory_push_item(self, leafdir, metadata, entry):
```

Comment:
I think this is not right, as the handles_type decorator was designed to be applied onto a method which is called once for each file in the leafdir, not once for the entire leafdir.  That's why my prototype didn't do it this way.

I believe it only seems to work right now in the test because simple_directories/dest1/RAW contains exactly one file, and because `__mixin_push_items` in the base class does not recursively walk the directory.  So it'll see exactly one file and you'll get exactly one DirectoryPushItem back.  If you adjust the testcase so the RAW directory contains more than one file at the top level, I think you'll start to see some duplicate DirectoryPushItem objects yielded.

Also, the case of a completely empty RAW directory should still generate a DirectoryPushItem.